File: FACULDADE/algoritmos/PYTHON/LISTA02/questao3.py

class Questao3:

    def __init__(self):
       self.lados = []

       for i in range(3):
            lado = float(input(f"Digite o lador do lado {i + 1}: "))
            if self.ehvalido(lado):
               self.lados.append(lado)
            else:
                self.lados.clear()
                self.__init__()
                break

    def ehvalido(self, lado):
        if lado > 0:
            return True
        else:
            return False

    def ehtriangulo(self):
        soma = 0
        for i in range(len(self.lados)):
            for j in range(len(self.lados)):
                if i != j:
                    soma += self.lados[j]
            if self.lados[i] > soma:
                return False
            soma = 0
        return True

    def __str__(self):
        if self.ehtriangulo():
            return (f"Os valores constituem um triângulo")
        else:
            return "Os Valores não constituem um triângulo"

if __name__ == '__main__':


    print(Questao3())

File: FACULDADE/algoritmos/C/Lista 2/questao3.c
#include <stdio.h>
#include <locale.h>



void apresentacao() 
{
	printf("Questão 3:\n");
	printf("Dado três valores, verificar se eles podem constituir os lados de um triângulo (triângulo é uma"
		"figura geométrica onde cada lado é menor do que a soma dos outros dois lados).\n");
}

void validalado(int status, float* lado, int i) 
{
	int temp, input;
	while (status != 1) 
	{
		while ((temp = getchar()) != EOF && temp != '\n');
		printf("Input inválido... \nDigite o valor do lado %d: ", i + 1);
		status = scanf("%f", lado);
	}
}

void pegalados(float lados[3]) 
{
	float lado;
	int status;
		for (int i = 0; i < 3; i++) 
		{
			printf("Digite o valor do lado %d: ", i + 1);
			status = scanf("%f", &lado);
			validalado(status, &lado, i);
			lados[i] = lado;
		}	
}
int ehtriangulo(float lados[3]) 
{
	float soma = 0;
	for (int i = 0; i < 3; i++) 
	{
		for (int j = 0; j < 3; j++) 
		{
			if (i != j) 
			{
				soma += lados[j];
			}
		}
		if (lados[i] > soma) return 0;

		soma = 0;
	}
	return 1;	
}

void imprimevalor(float lados[3]) 
{
	if (ehtriangulo(lados)) printf("Pode constituir os lados de um triângulo.\n");
	else printf("Não pode constituir os lados de um trângulo.\n");
}

int main() 
{
	setlocale(LC_ALL, "");
	float lados[3];

	apresentacao();
	pegalados(lados);
	imprimevalor(lados);
}